### ISSUE TITLE
Source interface for syslog is required when using TCP

### ISSUE THREAD

**Issue by adietrich-ussignal** [aristanetworks/ansible-avd#1595]
### Issue Summary

When specifying syslog configurations in eos_cli_config_gen, source-interface is required even when using TCP. This generates an inordinate amount of log messages that the configuration is not supported. 

When the source-interface is not provided, ansible-avd fails to successfully run the playbook. 

Per Arista documentation, "Configuring logging source-interface with a TCP host is not supported.  The source-interface option will be ignored."

### Which component(s) of AVD impacted

eos_cli_config_gen

### How do you run AVD ?

Ansible CLI with AVD Runner

### Steps to reproduce

```shell
1. Set logging protocol to TCP for syslog hosts with the source_interface specified.

logging:
  vrfs:
    management:
      source_interface: Management1
      hosts:
        syslog.host.1:
          protocol: tcp
        syslog.host.2:
          protocol: tcp

2. Push configuration to switches.

3. Inspect log message output for "%LOGMGR-4-TCP_SOURCE_INTERFACE_UNSUPPORTED" message.
```

### Relevant log output

```shell
%LOGMGR-4-TCP_SOURCE_INTERFACE_UNSUPPORTED: Source-Interface spoofing requested with TCP.

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'dict object' has no attribute 'source_interface'
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by ClausHolbechArista** [comment]
Thank you for reporting this. Let me take a look right away.

**Comment by ClausHolbechArista** [comment]
@adietrich-ussignal We have fixed this issue in the `devel` branch. The fix will be included in the next minor release v3.4.0, scheduled for end of March.

**Comment by adietrich-ussignal** [comment]
@ClausHolbechArista after testing with the newest release, it does not appear that the fix generates a valid configuration that CVP can handle. 

The only way I can get the ansible-avd playbook to complete successfully is by specifying this:
```
vrfs:
    management:
        source_interface: null
```

This appears to generate the following line of code in the configuration:
`logging source-interface None`

This subsequently causes CVP to mark this configuration line as an "error". 

I believe the source-interface null should not generate a source-interface line in the configuration, which should eliminate this issue.
